src/Work/ClassWork.tsx

```diff
@@ -111,8 +112,8 @@ export class ClassWork extends Work {
     return false;
   }
 
-  finish(): void {
-    if (!this.singularity) {
+  finish(cancelled: boolean, suppressDialog?: boolean): void {
+    if (!this.singularity && !suppressDialog && Settings.SuppressAugWorkFinishedPopup) {
       dialogBoxCreate(
```

Comment:
Here Settings.SupressAugWorkFinishedPopup being `true` is a requirement for actually creating the dialog box, this seems backwards.